core/src/main/resources/error/error-classes.json

```diff
@@ -11,6 +15,18 @@
     "message" : [ "%s cannot be represented as Decimal(%s, %s)." ],
     "sqlState" : "22005"
   },
+  "CANNOT_CREATE_OBJECT_DUE_TO_FAILED_DIRECTORY_CREATION" : {
+    "message" : [ "Unable to CREATE %s %s as failed to create its directory %s." ]
+  },
+  "CANNOT_CREATE_PARTITION" : {
```

Comment:
"partition" is pretty overloaded in Spark, maybe we should make this `CANNOT_CREATE_PARTITION_PATH`. Then we should replace it for `CANNOT_DELETE PARTITION` and so on too.

Reply:
cc @karenfeng , Could you please help me review it again? Thank you!